Analysis/Tasks/PWGHF/HFCandidateCreator3Prong.cxx

```diff
@@ -170,18 +182,38 @@ struct HFCandidateCreator3ProngMC {
       // Λc± → p± K∓ π±
       if (result == N3ProngDecays) {
         //Printf("Checking Λc± → p± K∓ π±");
-        if (RecoDecay::getMatchedMCRec(particlesMC, std::move(arrayDaughters), 4122, array{+kProton, -kKPlus, +kPiPlus}, true, &sign) > -1) {
+        auto indexRecLc = RecoDecay::getMatchedMCRec(particlesMC, std::move(arrayDaughters), 4122, array{+kProton, -kKPlus, +kPiPlus}, true, &sign, -1);
+        if (indexRecLc > -1) {
           result = sign * LcToPKPi;
+
+          //Printf("Flagging the different Λc± → p± K∓ π± decay channels");
+          RecoDecay::getDaughters(particlesMC, indexRecLc, &arrDaughIndex, array{+kProton, -kKPlus, +kPiPlus}, 1);
+          size = arrDaughIndex.size();
+          for (auto iProng = 0; iProng < size; ++iProng) {
+            auto daughI = particlesMC.iteratorAt(arrDaughIndex[iProng]);
+            arrPDGDaugh[iProng] = daughI.pdgCode();
```

Comment:
Why is this part executed regardless of the number of direct daughters? Shouldn't everything below `getDaughters` be executed only for the resonant decays, i.e. `if (arrDaughIndex.size() == 2)`?